oak_runtime/src/tests.rs

```diff
@@ -440,80 +454,9 @@ fn create_node_invalid_configuration_err() {
     );
 }
 
-/// Create a test Node that creates a Node with a less confidential label and fails.
-///
-/// If this succeeded, it would be a violation of information flow control, since the original
-/// confidential Node would be able to spawn "less confidential / public" Nodes and use their side
-/// effects as a covert channel to exfiltrate confidential data.
-#[test]
-fn create_node_less_confidential_label_err() {
-    let tag_0 = oak_abi::label::authorization_bearer_token_hmac_tag(&[1, 1, 1]);
-    let initial_label = Label {
-        confidentiality_tags: vec![tag_0],
-        integrity_tags: vec![],
-    };
-    let less_confidential_label = Label {
-        confidentiality_tags: vec![],
-        integrity_tags: vec![],
-    };
-    let initial_label_clone = initial_label.clone();
-    run_node_body(
-        &initial_label,
-        &NodePrivilege::default(),
-        Box::new(move |runtime| {
-            let (_write_handle, read_handle) = runtime.channel_create("", &initial_label_clone)?;
-            let node_configuration = NodeConfiguration {
-                config_type: Some(ConfigType::LogConfig(LogConfiguration {})),
-            };
-            let result = runtime.node_create(
-                "test",
-                &node_configuration,
-                &less_confidential_label,
-                read_handle,
-            );
-            assert_eq!(Err(OakStatus::ErrPermissionDenied), result);
-            Ok(())
-        }),
-    );
-}
-
-/// Create a test Node that creates a Node with a more confidential label and succeeds.
-#[test]
-fn create_node_more_confidential_label_ok() {
```

Comment:
I think it could be possible to fix this test to cover a case we are not covering (although the name would probably need to change). The initial node can be changed to also have a public label, can be given the privilege to declassify tag_0 so that the channel creation can succeed.

I think a gRPC client node can be used to test creation of a non-public node, where the URI of the gRPC client configuration and the authority of the TLS endpoint tag line up. The gRPC client node will not try to connect to the external service until it receives an invocation, so it should be fine to use it without an actual service listening.